### PR TITLE
ng_pktbuf: Some fixes

### LINKED ISSUE #3088: pktbuf: compiler warning on atomic_dec()
Compiling `ng_pktbuf` since 6821350 yields a compiler warning.

```
/home/martine/Repositories/RIOT-OS/RIOT/sys/net/crosslayer/ng_pktbuf/ng_pktbuf.c: In function 'ng_pktbuf_start_write':
/home/martine/Repositories/RIOT-OS/RIOT/sys/net/crosslayer/ng_pktbuf/ng_pktbuf.c:151:20: warning: passing argument 1 of 'atomic_dec' from incompatible pointer type
         atomic_dec(&pkt->users);
                    ^
In file included from /home/martine/Repositories/RIOT-OS/RIOT/core/include/mutex.h:25:0,
                 from /home/martine/Repositories/RIOT-OS/RIOT/sys/net/crosslayer/ng_pktbuf/ng_pktbuf.c:25:
/home/martine/Repositories/RIOT-OS/RIOT/core/include/atomic.h:81:19: note: expected 'struct atomic_int_t *' but argument is of type 'unsigned int *'
 static inline int atomic_dec(atomic_int_t *var)
                   ^
```

However, `atomic_int_t`'s member is signed while `pkt->users` is unsigned, which could make simple casting problematic (not to mention that `atomic_int_t::value` is volatile).

### PR DESCRIPTION
**Pull request by authmillenon**
Fixes #3088 and also some weird artifacts I discovered when writing #3089.

### CODE REVIEW

File: sys/net/crosslayer/ng_pktbuf/ng_pktbuf.c [hunk 1]
```diff
@@ -249,28 +252,12 @@ static ng_pktsnip_t *_pktbuf_duplicate(const ng_pktsnip_t *pkt)
...
-
-                _pktbuf_internal_free(res->data);
-                _pktbuf_internal_free(res);
-
-                res = next;
-            } while (res);
-
-            return NULL;
-        }
-    }
```
Comment (authmillenon) [inline sys/net/crosslayer/ng_pktbuf/ng_pktbuf.c]: I'm a little unsure about this removal myself. On the one hand it makes total sense in the concept of the packet buffer, but I'm not sure why it was here in the first place. So please check your favorite network stack apps thoroughly if they still work with this PR

Comment (authmillenon) [inline sys/net/crosslayer/ng_pktbuf/ng_pktbuf.c]: Yes, I am really confused about this while-loop

Comment (OlegHahm) [inline sys/net/crosslayer/ng_pktbuf/ng_pktbuf.c]: For clarification: the current code duplicates the whole packet, while this PR would result in duplicating only the current sniplet itself, right?

File: sys/net/crosslayer/ng_pktbuf/ng_pktbuf.c [hunk 2]
```diff
@@ -30,6 +30,9 @@
 
 #include "_pktbuf_internal.h"
 
+#define ENABLE_DEBUG (0)
+#include "debug.h"
```
Comment (OlegHahm) [inline sys/net/crosslayer/ng_pktbuf/ng_pktbuf.c]: This overlaps with #3089.

Comment (authmillenon) [inline sys/net/crosslayer/ng_pktbuf/ng_pktbuf.c]: I know, but the change to _duplicate also generates a merge conflict if I would add the DEBUG output of it to#3089. So I added it here. Afaik git should be able to handle this kind of conflict here.

File: sys/net/crosslayer/ng_pktbuf/ng_pktbuf.c (hunk 1 above)
Comment (authmillenon) [inline sys/net/crosslayer/ng_pktbuf/ng_pktbuf.c]: I actually don't know. It certainly looks like it, but I thought I removed this ages ago...

Comment (BytesGalore) [inline sys/net/crosslayer/ng_pktbuf/ng_pktbuf.c]: I'm currently using this PR in conjunction with #2895 and it does not misbehave, i.e. same bahaviour as without this PR.

Comment (authmillenon) [inline sys/net/crosslayer/ng_pktbuf/ng_pktbuf.c]: :+1:

Comment (authmillenon) [inline sys/net/crosslayer/ng_pktbuf/ng_pktbuf.c]: (just hopefully with a little bit more memory efficient packet buffer)

**Comment by OlegHahm** [comment]
ACK

**Comment by authmillenon** [comment]
And go.

**Pull request merged**